File: benchmarks/tsb/bench_sort_ops.ts
/**
 * Benchmark: sortValuesSeries and sortValuesDataFrame on 100k rows
 */
import { Series, DataFrame, sortValuesSeries, sortValuesDataFrame } from "../../src/index.ts";

const ROWS = 100_000;
const WARMUP = 3;
const ITERATIONS = 10;

const data = Array.from({ length: ROWS }, (_, i) => Math.sin(i) * 1000);
const s = new Series({ data });

const dfData = {
  a: Array.from({ length: ROWS }, (_, i) => Math.sin(i) * 1000),
  b: Array.from({ length: ROWS }, (_, i) => Math.cos(i) * 500),
};
const df = new DataFrame(dfData);

for (let i = 0; i < WARMUP; i++) {
  sortValuesSeries(s);
  sortValuesDataFrame(df, "a");
}

const start = performance.now();
for (let i = 0; i < ITERATIONS; i++) {
  sortValuesSeries(s);
  sortValuesDataFrame(df, "a");
}
const total = performance.now() - start;

console.log(
  JSON.stringify({
    function: "sort_ops",
    mean_ms: total / ITERATIONS,
    iterations: ITERATIONS,
    total_ms: total,
  }),
);

File: benchmarks/pandas/bench_sort_ops.py
"""Benchmark: Series.sort_values and DataFrame.sort_values on 100k rows"""
import json, time
import numpy as np
import pandas as pd

ROWS = 100_000
WARMUP = 3
ITERATIONS = 10

data = np.sin(np.arange(ROWS, dtype=float)) * 1000
s = pd.Series(data)
df = pd.DataFrame({
    "a": np.sin(np.arange(ROWS, dtype=float)) * 1000,
    "b": np.cos(np.arange(ROWS, dtype=float)) * 500,
})

for _ in range(WARMUP):
    s.sort_values()
    df.sort_values("a")

start = time.perf_counter()
for _ in range(ITERATIONS):
    s.sort_values()
    df.sort_values("a")
total = (time.perf_counter() - start) * 1000

print(json.dumps({
    "function": "sort_ops",
    "mean_ms": total / ITERATIONS,
    "iterations": ITERATIONS,
    "total_ms": total,
}))
